neo4j/example/src/test/java/example/springdata/neo4j/ActorRepositoryIntegrationTest.java

```diff
@@ -57,7 +60,54 @@ public void shouldBeAbleToSaveAndLoadActor() {
 		assertThat(actorRepository.findById(daniel.getId())).hasValueSatisfying(actor -> {
 			assertThat(actor.getName()).isEqualTo(daniel.getName());
 			assertThat(actor.getRoles()).hasSize(1).first()
-					.satisfies(role -> assertThat(role.getRole()).isEqualTo("Harry Potter"));
+				.satisfies(role -> assertThat(role.getRole()).isEqualTo("Harry Potter"));
 		});
 	}
+
+	@Test // SEE #386
+	public void shouldBeAbleToHandleNestedProperties() {
+
+		Assume.assumeTrue(thatSupportForNestedPropertiesIsAvailable());
+
+		Movie theParentTrap = new Movie("The Parent Trap");
+		Movie iKnowWhoKilledMe = new Movie("I Know Who Killed Me");
+
+		Actor lindsayLohan = new Actor("Lindsay Lohan");
+
+		lindsayLohan.actedIn(theParentTrap, "Hallie Parker");
+		lindsayLohan.actedIn(theParentTrap, "Annie James");
+		lindsayLohan.actedIn(iKnowWhoKilledMe, "Aubrey Fleming");
+		lindsayLohan.actedIn(iKnowWhoKilledMe, "Dakota Moss");
+		actorRepository.save(lindsayLohan);
+
+		Actor nealMcDonough = new Actor("Neal McDonough");
+		nealMcDonough.actedIn(iKnowWhoKilledMe, "Daniel Fleming");
+		actorRepository.save(nealMcDonough);
+
+		assertThat(actorRepository.findAllByRolesMovieTitle(iKnowWhoKilledMe.getTitle())).hasSize(2)
+			.extracting(Actor::getName).contains(lindsayLohan.getName(), nealMcDonough.getName());
+	}
+
+	private static boolean thatSupportForNestedPropertiesIsAvailable() {
+
+		Optional<String> version = Optional.ofNullable(SpringBootVersion.getVersion());
+		return version.map(v -> v.split("\\."))
+			.filter(c -> c.length >= 3)
+			.map(v -> new Integer[] { Integer.valueOf(v[0]), Integer.valueOf(v[1]), Integer.valueOf(v[2]) })
+			.map(c -> c[0] >= 2 && (c[1] >= 1 || c[1] == 0 && c[2] >= 5))
+			.orElseGet(ActorRepositoryIntegrationTest::boot210Or205SpecificMethodExists);
+	}
+
+	private static boolean boot210Or205SpecificMethodExists() {
```

Comment:
Shall we rather use `SpringBootVersion` in combination with Spring Data's `Version` here?

Reply:
We do. But `SpringBootVersion` explicitly states:

> Note that some ClassLoaders do not expose the package metadata, hence this class might not be able to determine the Spring Boot version in all environments. Consider using a reflection-based check instead.